src/outline/manager.js

```diff
@@ -442,16 +442,32 @@ export default function (context) {
         console.error('Unable to load profile', err)
         return []
       }
-      const pods = kb.each(me, ns.space('storage'), null, me.doc())
+      // load pod's storages from profile
+      let pods = kb.each(me, ns.space('storage'), null, me.doc())
+      pods.map(async (pod) => {
+        // TODO use addPodStorageFromUrl(pod.uri) to check for pim:Storage ???
+        await loadContainerRepresentation(pod)
+      })
 
       try {
         // if uri then SolidOS is a browse.html web app
         const uri = (new URL(window.location.href)).searchParams.get('uri')
         const podUrl = uri || window.location.href
-        await addPodRootFromUrl(podUrl.substring(0, podUrl.lastIndexOf('/') + 1))
+        await addPodStorageFromUrl(podUrl)
       } catch (err) {
         console.error('cannot load container', err)
       }
+      // remove namedNodes duplicates
+      function uniques (nodes) {
+        const uniqueNodes = []
+        nodes.forEach(node => {
+          if (!uniqueNodes.find(uniqueNode => uniqueNode.equals(node))) uniqueNodes.push(node)
+        })
+        return uniqueNodes
+      }
+      pods = uniques(pods)
+      console.log('return pods')
```

Comment:
suggestion: if it's not too much trouble I would use debug instead of console.log.